configuration/price.go

```
@@ -0,0 +1,54 @@
+package configuration
+
+type PricesInfo struct {
+	WeekDayHourlyPrice    float32
+	HoursWeekDay          int
+	WeekendDayHourlyPrice float32
+	HoursWeekendDay       int
+	BhDayHourlyPrice      float32
+	HoursBhDay            int
+}
+
+func (c *Configuration) GetPricesInfo() (*PricesInfo, error) {
+	weekDayPrice, err := c.FindPriceByDay("weekday")
+	if err != nil {
+		return nil, err
+	}
+	excludedWeekDayHoursAmount := 0
+	excludedHours, _ := c.FindRotationExcludedHoursByDay("weekday")
+	if excludedHours != nil {
+		excludedWeekDayHoursAmount = excludedHours.ExcludedEndsAt - excludedHours.ExcludedStartsAt
+	}
+	weekDayWorkingHours := 24 - excludedWeekDayHoursAmount
+
+	weekendDayPrice, err := c.FindPriceByDay("weekend")
+	if err != nil {
+		return nil, err
+	}
+	excludedWeekendDayHoursAmount := 0
+	excludedHours, _ = c.FindRotationExcludedHoursByDay("weekend")
+	if excludedHours != nil {
+		excludedWeekendDayHoursAmount = excludedHours.ExcludedEndsAt - excludedHours.ExcludedStartsAt
+	}
+	weekendDayWorkingHours := 24 - excludedWeekendDayHoursAmount
+
+	bhDayPrice, err := c.FindPriceByDay("bankholiday")
+	if err != nil {
+		return nil, err
+	}
+	excludedBhDayHoursAmount := 0
+	excludedHours, _ = c.FindRotationExcludedHoursByDay("bankholiday")
```

again, why err is discarded?

---

see https://github.com/form3tech-oss/go-pagerduty-oncall-report/pull/41#discussion_r971693406

---

see https://github.com/form3tech-oss/go-pagerduty-oncall-report/pull/41#discussion_r971695312

---

Adding err handling won't make it worse. I would personally add err handling here.

---

https://github.com/form3tech-oss/go-pagerduty-oncall-report/pull/41#discussion_r971892679